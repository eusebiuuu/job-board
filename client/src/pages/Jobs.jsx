import { useEffect } from "react"
import styles from './Jobs.module.css'
import Filters from "../components/Filters";
import JobCard from "../components/JobCard";
import { useDispatch, useSelector } from "react-redux";
import { nanoid } from "nanoid";
import { getAllJobs } from "../redux/jobs/jobsSlice";
import Loader from "../components/Loader";

export default function Jobs() {
  const { isLoading, filteredJobs } = useSelector((state) => state.jobs);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAllJobs());
    // eslint-disable-next-line
  }, []);

  return (<div className={styles.container}>
    {isLoading || !filteredJobs
      ? <Loader />
      : <>
          <Filters show={true} />
          <div className={styles.title}>
          {filteredJobs.length !== 0
            ? <h2>Found jobs</h2>
            : <h2>No jobs found...</h2>}
          </div>
          <div className={styles.jobs}>
            {filteredJobs.map(job => {
              return <JobCard key={nanoid()} apply job={job} />
            })}
          </div>
        </>
    }
  </div>)
}
