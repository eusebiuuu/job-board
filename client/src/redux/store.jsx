import { configureStore } from '@reduxjs/toolkit';
import jobsSlice from './jobs/jobsSlice';
import candidateSlice from './candidate/candidateSlice';
import companySlice from './company/companySlice';

export const store = configureStore({
  reducer: {
    jobs: jobsSlice,
    candidate: candidateSlice,
    company: companySlice,
  },
});