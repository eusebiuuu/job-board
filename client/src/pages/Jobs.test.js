import { render } from '@testing-library/react';
import Jobs from './Jobs';
import { Provider } from 'react-redux';
import { store } from '../redux/store';

it('test if Jobs component renders properly', () => {
  expect(render(<Provider store={store}>
    <Jobs />
  </Provider>)).toMatchSnapshot();
})