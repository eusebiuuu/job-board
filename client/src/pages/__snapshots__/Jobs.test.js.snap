// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if Jobs component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          aria-busy="true"
          aria-label="oval-loading"
          data-testid="oval-loading"
          role="status"
          style="display: flex; padding: 3px;"
        >
          <svg
            data-testid="oval-svg"
            height="100"
            stroke="#222"
            viewBox="-20 -20 42 42"
            width="100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                data-testid="oval-secondary-group"
                stroke-width="2"
                transform="translate(1 1)"
              >
                <circle
                  cx="0"
                  cy="0"
                  r="20"
                  stroke="#fff"
                  stroke-opacity=".5"
                  stroke-width="2"
                />
                <path
                  d="M20 0c0-9.94-8.06-20-20-20"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="1s"
                    from="0 0 0"
                    repeatCount="indefinite"
                    to="360 0 0"
                    type="rotate"
                  />
                </path>
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        aria-busy="true"
        aria-label="oval-loading"
        data-testid="oval-loading"
        role="status"
        style="display: flex; padding: 3px;"
      >
        <svg
          data-testid="oval-svg"
          height="100"
          stroke="#222"
          viewBox="-20 -20 42 42"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              data-testid="oval-secondary-group"
              stroke-width="2"
              transform="translate(1 1)"
            >
              <circle
                cx="0"
                cy="0"
                r="20"
                stroke="#fff"
                stroke-opacity=".5"
                stroke-width="2"
              />
              <path
                d="M20 0c0-9.94-8.06-20-20-20"
              >
                <animatetransform
                  attributeName="transform"
                  dur="1s"
                  from="0 0 0"
                  repeatCount="indefinite"
                  to="360 0 0"
                  type="rotate"
                />
              </path>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
